### PR TITLE
MLPAB-690 Send an email once signed, unless the CP hasn't started

### PR DESCRIPTION
**Pull request by hawx**

### CODE REVIEW

File: app/internal/page/share_code.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+func NewShareCodeSender(dataStore DataStore, notifyClient NotifyClient, appPublicURL string, randomString func(int) string) *shareCodeSender {
+	return &shareCodeSender{
+		dataStore:    dataStore,
+		notifyClient: notifyClient,
+		appPublicURL: appPublicURL,
+		randomString: randomString,
+	}
+}
+
+func (s *shareCodeSender) Send(ctx context.Context, template notify.TemplateId, appData AppData, email string) error {
```
Comment (hawx) [inline app/internal/page/share_code.go:31]: I hate having a Xer.X but it has cleaned up the code

Comment (acsauk) [reply app/internal/page/share_code.go:31]: It's a small price to pay

File: app/internal/page/share_code_test.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+func (m *mockNotifyClient) Email(ctx context.Context, email notify.Email) (string, error) {
+	args := m.Called(ctx, email)
+	return args.String(0), args.Error(1)
+}
+
+func (m *mockNotifyClient) Sms(ctx context.Context, sms notify.Sms) (string, error) {
+	args := m.Called(ctx, sms)
+	return args.String(0), args.Error(1)
+}
```
Comment (acsauk) [inline app/internal/page/share_code_test.go:30]: Would you hate this living in `page/mock_test`?

Comment (hawx) [reply app/internal/page/share_code_test.go:30]: not at all

**Review by acsauk — APPROVED** [review]
:shipit:

**Pull request merged**